mswindows/osgeo4w/package.sh

```diff
@@ -41,55 +41,69 @@ else
 fi
 
 export OSGEO4W_ROOT_MSYS="/c/OSGeo4W${OSGEO4W_POSTFIX}"
-export OSGEO4W_ROOT="C:\\\OSGeo4W${OSGEO4W_POSTFIX}"
-
-export PATH=/usr/bin:/mingw${MINGW_POSTFIX}/bin/:$OSGEO4W_ROOT_MSYS/bin:$PWD/mswindows/osgeo4w/lib:$PWD/mswindows/osgeo4w
-
-# Force Python3
-# alias python=python3
-export PYTHONPATH=
-export PYTHONHOME=$OSGEO4W_ROOT_MSYS/apps/Python37
-export PATH=$OSGEO4W_ROOT_MSYS/apps/Python37:$PATH
-
-T0=$(date +%s) 
-LT=$T0 
-CS="" 
-
-log() { 
-    local D T 
-    NOW=$(date) 
-    T=$(date +%s) 
-    
-    if [ -n "$CS" ]; then 
-        local D H M S 
- 	S=$(( $T-$LT )) 
- 	M=$(( S/60 )); S=$(( S%60 )) 
- 	H=$(( M/60 )); M=$(( M%60 )) 
- 	D=$(( H/24 )); H=$(( H%24 )) 
- 	
- 	echo -n "$NOW: FINISHED $CS AFTER " 
- 	(( D>0 )) && echo -n "${D}d" 
- 	(( H>0 )) && echo -n "${H}h" 
- 	(( M>0 )) && echo -n "${M}m" 
- 	echo "${S}s" 
-    fi 
-    
-    CS="$@" 
-    LT=$T 
-    if [ -n "$CS" ]; then 
-        echo $NOW: STARTING $CS 
-    elif [ -n "$T0" ]; then 
- 	CS="COMPLETE RUN" 
- 	LT=$T0 
- 	T0="" 
- 	log 
-    fi 
+export OSGEO4W_ROOT=$(cygpath -w "$OSGEO4W_ROOT_MSYS")
+
+fetchenv() {
+    local IFS
+    IFS=
+    batch=$1
+    shift
+    srcenv=$(mktemp /tmp/srcenv.XXXXXXXXXX)
+    dstenv=$(mktemp /tmp/dstenv.XXXXXXXXXX)
+    diffenv=$(mktemp /tmp/diffenv.XXXXXXXXXX)
+    args="$@"
+    cmd.exe /c set >$srcenv
```

Comment:
tested on the OSGeo4W shell:

```
C:\>cmd.exe /c set
ALLUSERSPROFILE=C:\ProgramData
APPDATA=C:\Users\hkmyr\AppData\Roaming
CommonProgramFiles=C:\Program Files\Common Files
CommonProgramFiles(x86)=C:\Program Files (x86)\Common Files
CommonProgramW6432=C:\Program Files\Common Files
COMPUTERNAME=mylaptop
ComSpec=C:\WINDOWS\system32\cmd.exe
CYGWIN=nodosfilewarning
DriverData=C:\Windows\System32\Drivers\DriverData
[...]
```
it works; though testing in msys2/ming64 shell
```

$ bash --version
GNU bash, version 4.4.23(1)-release (x86_64-pc-msys)

user@mylaptop MINGW64 ~
$ cmd.exe /c set
Microsoft Windows [Version 10.0.18362.356]
(c) 2019 Microsoft Corporation. Alle Rechte vorbehalten.

C:\msys64\home\user>

C:\msys64\home\user>exit
exit

user@mylaptop MINGW64 ~
$
```
it seems there is no output of cmd.exe /c set.

anything missing here?

Reply:
odd.

```
jef@faramir MINGW64 ~
$ echo $PATH; file env.sh; cat env.sh; echo; bash env.sh
/mingw64/bin:/usr/local/bin:/usr/bin:/bin:/c/Windows/System32:/c/Windows:/c/Windows/System32/Wbem:/c/Windows/System32/WindowsPowerShell/v1.0/:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl
env.sh: ASCII text
type -a cmd.exe
cmd.exe /c set >1
cmd.exe /c "call `cygpath -w /c/OSGeo4W64/bin/o4w_env.bat` \>nul 2\>nul \& set" >2
ls -l 1 2

cmd.exe is /c/Windows/System32/cmd.exe
-rw-r--r-- 1 jef NORBIT+Group(513) 4312 Sep 24 23:36 1
-rw-r--r-- 1 jef NORBIT+Group(513) 4823 Sep 24 23:36 2
```

Reply:
any news?

Reply:
testing in MSYS2/MSYS:

```
----

mylab@DESKTOP-VADT8Q4 MSYS ~
$ echo $PATH
/usr/local/bin:/usr/bin:/bin:/opt/bin:/c/Windows/System32:/c/Windows:/c/Windows/System32/Wbem:/c/Windows/System32/WindowsPowerShell/v1.0/:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl

----

mylab@DESKTOP-VADT8Q4 MSYS ~
$ type -a cmd.exe
cmd.exe is /c/Windows/System32/cmd.exe

----

mylab@DESKTOP-VADT8Q4 MSYS ~
$ cmd.exe /c set
Microsoft Windows [Version 10.0.18362.418]
(c) 2019 Microsoft Corporation. Alle Rechte vorbehalten.

C:\msys64\home\mylab>exit
exit

----

mylab@DESKTOP-VADT8Q4 MSYS ~
$ cmd.exe /c set >1


exit

----

```

and MSYS2/MinGW 64-bit:

```
----

mylab@DESKTOP-VADT8Q4 MINGW64 ~
$ echo $PATH
/mingw64/bin:/usr/local/bin:/usr/bin:/bin:/c/Windows/System32:/c/Windows:/c/Windows/System32/Wbem:/c/Windows/System32/WindowsPowerShell/v1.0/:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl

----

mylab@DESKTOP-VADT8Q4 MINGW64 ~
$ type -a cmd.exe
cmd.exe is /c/Windows/System32/cmd.exe

----

mylab@DESKTOP-VADT8Q4 MINGW64 ~
$ cmd.exe /c set
Microsoft Windows [Version 10.0.18362.418]
(c) 2019 Microsoft Corporation. Alle Rechte vorbehalten.

C:\msys64\home\mylab>exit
exit

----

mylab@DESKTOP-VADT8Q4 MINGW64 ~
$ cmd.exe /c set >1

exit

----
```

cmd.exe and $PATH is the same like in your environment.

it seems _cmd.exe /c set_ is switching into the cmd shell, but doesn't give the output as it does e.g. in the OSGeo4W command line:


```
C:\>where cmd.exe
C:\Windows\System32\cmd.exe

C:\>cmd.exe /c set
ALLUSERSPROFILE=C:\ProgramData
APPDATA=C:\Users\mylab\AppData\Roaming
CommonProgramFiles=C:\Program Files\Common Files
CommonProgramFiles(x86)=C:\Program Files (x86)\Common Files
CommonProgramW6432=C:\Program Files\Common Files
[...]
```

any idea?

Reply:
@hellik Does `//c` instead of `/c` work for you?

Reply:
tested it now on another win 10 box, not exactly in MSYS2/MinGW 64-bit, but in another MinGW64 environment.

```
mylaptop@ MINGW64 ~
$ cmd.exe /c set
ACLOCAL_PATH=C:\Program Files\Git\mingw64\share\aclocal;C:\Program Files\Git\usr\share\aclocal
[...]
COMSPEC=C:\windows\system32\cmd.exe
[...]
```

```
mylaptop@ MINGW64 ~
$ cmd.exe //c set
ACLOCAL_PATH=C:\Program Files\Git\mingw64\share\aclocal;C:\Program Files\Git\usr\share\aclocal
[...]
COMSPEC=C:\windows\system32\cmd.exe
[...]
```

both are working here.

it's odd as it seems there be several win 10 flavors with different behaviour. will test _cmd.exe //c set_ later on in my winGRASS build environment and report back.

Reply:
Apparently [MSYS' posix path conversion](http://www.mingw.org/wiki/Posix_path_conversion) in action - still odd.

Reply:
applied PR #137 locally

```
$ OSGEO4W_POSTFIX=64 ./mswindows/osgeo4w/package.sh
`C:/msys64/mingw64/bin/zlib1.dll' -> `dist.x86_64-w64-mingw32/bin/zlib1.dll'
`C:/msys64/mingw64/bin/libbz2-1.dll' -> `dist.x86_64-w64-mingw32/bin/libbz2-1.dll'
`C:/msys64/mingw64/bin/libiconv-2.dll' -> `dist.x86_64-w64-mingw32/bin/libiconv-2.dll'
`C:/msys64/mingw64/bin/libexpat-1.dll' -> `dist.x86_64-w64-mingw32/bin/libexpat-1.dll'
`C:/msys64/mingw64/bin/libfontconfig-1.dll' -> `dist.x86_64-w64-mingw32/bin/libfontconfig-1.dll'
`C:/msys64/mingw64/bin/libintl-8.dll' -> `dist.x86_64-w64-mingw32/bin/libintl-8.dll'
`C:/msys64/mingw64/bin/libsystre-0.dll' -> `dist.x86_64-w64-mingw32/bin/libsystre-0.dll'
`C:/msys64/mingw64/bin/libtre-5.dll' -> `dist.x86_64-w64-mingw32/bin/libtre-5.dll'
`C:/msys64/mingw64/bin/libwinpthread-1.dll' -> `dist.x86_64-w64-mingw32/bin/libwinpthread-1.dll'
`C:/msys64/mingw64/bin/libcairo-2.dll' -> `dist.x86_64-w64-mingw32/bin/libcairo-2.dll'
`C:/msys64/mingw64/bin/libpixman-1-0.dll' -> `dist.x86_64-w64-mingw32/bin/libpixman-1-0.dll'
`C:/msys64/mingw64/bin/libpng16-16.dll' -> `dist.x86_64-w64-mingw32/bin/libpng16-16.dll'
`C:/msys64/mingw64/bin/libfreetype-6.dll' -> `dist.x86_64-w64-mingw32/bin/libfreetype-6.dll'
`C:/msys64/mingw64/bin/libharfbuzz-0.dll' -> `dist.x86_64-w64-mingw32/bin/libharfbuzz-0.dll'
`C:/msys64/mingw64/bin/libglib-2.0-0.dll' -> `dist.x86_64-w64-mingw32/bin/libglib-2.0-0.dll'
`C:/msys64/mingw64/bin/libgraphite2.dll' -> `dist.x86_64-w64-mingw32/bin/libgraphite2.dll'
`C:/msys64/mingw64/bin/libpcre-1.dll' -> `dist.x86_64-w64-mingw32/bin/libpcre-1.dll'
`C:/msys64/mingw64/bin/libstdc++-6.dll' -> `dist.x86_64-w64-mingw32/bin/libstdc++-6.dll'
`C:/msys64/mingw64/bin/libgcc_s_seh-1.dll' -> `dist.x86_64-w64-mingw32/bin/libgcc_s_seh-1.dll'
`C:/OSGeo4W64/lib/libpq.lib' -> `mswindows/osgeo4w/lib/pq.lib'
Wed Oct 9 18:19:18 GMT 2019: STARTING remove old logs
Wed Oct  9 18:19:19 GMT 2019: FINISHED remove old logs AFTER 1s
Wed Oct 9 18:19:19 GMT 2019: STARTING configure
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
loading site script /mingw64/etc/config.site
checking host system type... config.sub: missing argument
Try `config.sub --help' for more information.

checking for gcc... gcc
checking whether the C compiler (gcc  ) works... yes
checking whether the C compiler (gcc  ) is a cross-compiler... no
checking whether we are using GNU C... yes
checking whether gcc accepts -g... yes
checking for Cygwin environment... no
checking for mingw32 environment... yes
checking for executable suffix... .exe
checking for full floating-point support... yes
checking for pwd... /C/OSGEO4~1/apps/msys/bin/pwd
checking for source directory... /c/msys64/usr/src/grass_win_fix/grassw
checking for build directory... /c/msys64/usr/src/grass_win_fix/grassw
checking for git... /C/msys64/usr/bin/git
checking for MacOSX App... no
checking for MacOSX architectures... no
checking for MacOSX SDK... no
checking how to build libraries... shared
configure: error: ***Unknown platform: ***
```

Reply:
taken from config.log

```
This file contains any messages produced by compilers while
running configure, to aid debugging if configure makes a mistake.

config.site:13: default build_alias set to x86_64-w64-mingw32
config.site:20: default prefix set to /mingw64
configure:766: checking host system type
configure:789: checking for gcc
configure:893: checking whether the C compiler (gcc  ) works
configure:909: gcc -o conftest    conftest.c  1>&5
configure:906:1: warning: return type defaults to 'int' [-Wimplicit-int]
  906 | main(){return(0);}
      | ^~~~
configure:935: checking whether the C compiler (gcc  ) is a cross-compiler
configure:940: checking whether we are using GNU C
configure:947: gcc -E conftest.c
configure:965: checking whether gcc accepts -g
configure:994: checking for Cygwin environment
configure:1008: gcc -c -g -O2  conftest.c 1>&5
configure: In function 'main':
configure:1002:20: error: '__CYGWIN32__' undeclared (first use in this function)
 1002 | #define __CYGWIN__ __CYGWIN32__
      |                    ^~~~~~~~~~~~
configure:1004:8: note: in expansion of macro '__CYGWIN__'
 1004 | return __CYGWIN__;
      |        ^~~~~~~~~~
configure:1002:20: note: each undeclared identifier is reported only once for each function it appears in
 1002 | #define __CYGWIN__ __CYGWIN32__
      |                    ^~~~~~~~~~~~
configure:1004:8: note: in expansion of macro '__CYGWIN__'
 1004 | return __CYGWIN__;
      |        ^~~~~~~~~~
configure: failed program was:
#line 997 "configure"
#include "confdefs.h"

int main() {

#ifndef __CYGWIN__
#define __CYGWIN__ __CYGWIN32__
#endif
return __CYGWIN__;
; return 0; }
configure:1024: checking for mingw32 environment
```

Reply:
> taken from config.log
> …

that's just successfully detecting that you are not running on cygwin, but doesn't give a hint why sed doesn't work for you.

Reply:
now changed in package.sh from

```
./configure \
	        --host=$conf_host \
	        [...]
```
to

```
	bash -x ./configure \
	        --host=$conf_host \
	        [...]
```
it seems to be an issue with proj detection:

from config.log:

```
configure:5698: checking for location of External PROJ includes
configure:5725: checking for proj.h
configure:5733: gcc -E  -IC:/OSGeo4W64/include   -IC:/OSGeo4W64/include conftest.c >/dev/null 2>conftest.out
+ eval '$CPP' '$CPPFLAGS' conftest.c '>/dev/null' '2>conftest.out'
++ gcc -E -IC:/OSGeo4W64/include -IC:/OSGeo4W64/include conftest.c
configure:5765: checking External PROJ major version
configure:5785: gcc -o conftest.exe -g -O2   -IC:/OSGeo4W64/include   -IC:/OSGeo4W64/include  -Wl,--export-dynamic,--enable-runtime-pseudo-reloc  -LC:\\OSGEO4~1/lib conftest.c  1>&5
+ eval '${CC-cc}' -o 'conftest${ac_exeext}' '$CFLAGS' '$CPPFLAGS' '$LDFLAGS' 'conftest.$ac_ext' '$LIBS' '1>&5'
++ gcc -o conftest.exe -g -O2 -IC:/OSGeo4W64/include -IC:/OSGeo4W64/include -Wl,--export-dynamic,--enable-runtime-pseudo-reloc '-LC:\\OSGEO4~1/lib' conftest.c
c:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: warning: --export-dynamic is not supported for PE+ targets, did you mean --export-all-symbols?
configure:5802: checking External PROJ minor version
configure:5822: gcc -o conftest.exe -g -O2   -IC:/OSGeo4W64/include   -IC:/OSGeo4W64/include  -Wl,--export-dynamic,--enable-runtime-pseudo-reloc  -LC:\\OSGEO4~1/lib conftest.c  1>&5
+ eval '${CC-cc}' -o 'conftest${ac_exeext}' '$CFLAGS' '$CPPFLAGS' '$LDFLAGS' 'conftest.$ac_ext' '$LIBS' '1>&5'
++ gcc -o conftest.exe -g -O2 -IC:/OSGeo4W64/include -IC:/OSGeo4W64/include -Wl,--export-dynamic,--enable-runtime-pseudo-reloc '-LC:\\OSGEO4~1/lib' conftest.c
c:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: warning: --export-dynamic is not supported for PE+ targets, did you mean --export-all-symbols?
configure:5839: checking External PROJ patch version
configure:5859: gcc -o conftest.exe -g -O2   -IC:/OSGeo4W64/include   -IC:/OSGeo4W64/include  -Wl,--export-dynamic,--enable-runtime-pseudo-reloc  -LC:\\OSGEO4~1/lib conftest.c  1>&5
+ eval '${CC-cc}' -o 'conftest${ac_exeext}' '$CFLAGS' '$CPPFLAGS' '$LDFLAGS' 'conftest.$ac_ext' '$LIBS' '1>&5'
++ gcc -o conftest.exe -g -O2 -IC:/OSGeo4W64/include -IC:/OSGeo4W64/include -Wl,--export-dynamic,--enable-runtime-pseudo-reloc '-LC:\\OSGEO4~1/lib' conftest.c
c:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: warning: --export-dynamic is not supported for PE+ targets, did you mean --export-all-symbols?
configure:5889: checking for location of External PROJ library
configure:5913: checking for proj_pj_info in -lproj
configure:5930: gcc -o conftest.exe -g -O2    -IC:/OSGeo4W64/include  -LC:/OSGeo4W64/lib  -Wl,--export-dynamic,--enable-runtime-pseudo-reloc  -LC:\\OSGEO4~1/lib conftest.c -lproj   1>&5
+ eval '${CC-cc}' -o 'conftest${ac_exeext}' '$CFLAGS' '$CPPFLAGS' '$LDFLAGS' 'conftest.$ac_ext' '$LIBS' '1>&5'
++ gcc -o conftest.exe -g -O2 -IC:/OSGeo4W64/include -LC:/OSGeo4W64/lib -Wl,--export-dynamic,--enable-runtime-pseudo-reloc '-LC:\\OSGEO4~1/lib' conftest.c -lproj
c:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: warning: --export-dynamic is not supported for PE+ targets, did you mean --export-all-symbols?
c:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lproj
collect2.exe: error: ld returned 1 exit status
configure: failed program was:
#line 5919 "configure"
#include "confdefs.h"
/* Override any gcc2 internal prototype to avoid an error.  */
/* We use char because int might match the return type of a gcc2
    builtin and then its argument prototype would still apply.  */
char proj_pj_info();

int main() {
proj_pj_info()
; return 0; }

```
related in package.log:
```

[...]
+ PROJINC=
+ PROJLIB=
+ PROJSHARE=
+ PROJ4API=0
+ echo -n 'checking for location of External PROJ includes... '
checking for location of External PROJ includes... + echo 'configure:5698: checking for location of External PROJ includes'
+ case "$with_proj_includes" in
+ echo C:/OSGeo4W64/include
C:/OSGeo4W64/include
+ test -n C:/OSGeo4W64/include
+ for dir in '$with_proj_includes'
+ test -d C:/OSGeo4W64/include
+ PROJINC=' -IC:/OSGeo4W64/include'
+ ac_save_cppflags='  -IC:/OSGeo4W64/include'
+ CPPFLAGS=' -IC:/OSGeo4W64/include   -IC:/OSGeo4W64/include'
+ for ac_hdr in proj.h
++ echo proj.h
++ sed y%./+-%__p_%
+ ac_safe=proj_h
+ echo -n 'checking for proj.h... '
checking for proj.h... + echo 'configure:5725: checking for proj.h'
+ cat
+ ac_try='$CPP $CPPFLAGS conftest.c >/dev/null 2>conftest.out'
+ eval echo configure:5733: '"$CPP' '$CPPFLAGS' conftest.c '>/dev/null' '2>conftest.out"'
++ echo configure:5733: 'gcc -E  -IC:/OSGeo4W64/include   -IC:/OSGeo4W64/include conftest.c >/dev/null 2>conftest.out'
++ grep -v '^ *+' conftest.out
++ grep -v '^conftest.c$'
+ ac_err=
+ test -z ''
+ rm -rf conftest.c conftest.out
+ eval ac_cv_header_proj_h=yes
++ ac_cv_header_proj_h=yes
+ rm -f 'conftest*'
++ echo '$ac_cv_header_proj_h'
+ eval 'test "$ac_cv_header_proj_h" = yes'
++ test yes = yes
+ echo yes
yes
++ echo proj.h
++ sed y%abcdefghijklmnopqrstuvwxyz./-%ABCDEFGHIJKLMNOPQRSTUVWXYZ___%
+ ac_tr_hdr=HAVE_PROJ_H
+ cat
+ CPPFLAGS='  -IC:/OSGeo4W64/include'
+ test 0 = 0
+ echo -n 'checking External PROJ major version... '
checking External PROJ major version... + echo 'configure:5765: checking External PROJ major version'
+ ac_save_cppflags='  -IC:/OSGeo4W64/include'
+ CPPFLAGS=' -IC:/OSGeo4W64/include   -IC:/OSGeo4W64/include'
+ test no = yes
+ cat
+ eval echo configure:5785: '"${CC-cc}' -o 'conftest${ac_exeext}' '$CFLAGS' '$CPPFLAGS' '$LDFLAGS' 'conftest.$ac_ext' '$LIBS' '1>&5"'
++ echo configure:5785: 'gcc -o conftest.exe -g -O2   -IC:/OSGeo4W64/include   -IC:/OSGeo4W64/include  -Wl,--export-dynamic,--enable-runtime-pseudo-reloc  -LC:\\OSGEO4~1/lib conftest.c  1>&5'
+ test -s conftest.exe
++ cat conftestdata
+ proj_ver_major=5
+ echo 5
5
+ rm -fr conftest.c conftest.exe conftestdata
+ CPPFLAGS='  -IC:/OSGeo4W64/include'
+ echo -n 'checking External PROJ minor version... '
checking External PROJ minor version... + echo 'configure:5802: checking External PROJ minor version'
+ ac_save_cppflags='  -IC:/OSGeo4W64/include'
+ CPPFLAGS=' -IC:/OSGeo4W64/include   -IC:/OSGeo4W64/include'
+ test no = yes
+ cat
+ eval echo configure:5822: '"${CC-cc}' -o 'conftest${ac_exeext}' '$CFLAGS' '$CPPFLAGS' '$LDFLAGS' 'conftest.$ac_ext' '$LIBS' '1>&5"'
++ echo configure:5822: 'gcc -o conftest.exe -g -O2   -IC:/OSGeo4W64/include   -IC:/OSGeo4W64/include  -Wl,--export-dynamic,--enable-runtime-pseudo-reloc  -LC:\\OSGEO4~1/lib conftest.c  1>&5'
+ test -s conftest.exe
++ cat conftestdata
+ proj_ver_minor=2
+ echo 2
2
+ rm -fr conftest.c conftest.exe conftestdata
+ CPPFLAGS='  -IC:/OSGeo4W64/include'
+ echo -n 'checking External PROJ patch version... '
checking External PROJ patch version... + echo 'configure:5839: checking External PROJ patch version'
+ ac_save_cppflags='  -IC:/OSGeo4W64/include'
+ CPPFLAGS=' -IC:/OSGeo4W64/include   -IC:/OSGeo4W64/include'
+ test no = yes
+ cat
+ eval echo configure:5859: '"${CC-cc}' -o 'conftest${ac_exeext}' '$CFLAGS' '$CPPFLAGS' '$LDFLAGS' 'conftest.$ac_ext' '$LIBS' '1>&5"'
++ echo configure:5859: 'gcc -o conftest.exe -g -O2   -IC:/OSGeo4W64/include   -IC:/OSGeo4W64/include  -Wl,--export-dynamic,--enable-runtime-pseudo-reloc  -LC:\\OSGEO4~1/lib conftest.c  1>&5'
+ test -s conftest.exe
++ cat conftestdata
+ proj_ver_patch=0
+ echo 0
0
+ rm -fr conftest.c conftest.exe conftestdata
+ CPPFLAGS='  -IC:/OSGeo4W64/include'
+ echo 'found PROJ version 5.2.0'
found PROJ version 5.2.0
+ test 5 = 5
++ expr 2 '<' 1
+ test 0 = 1
+ test 0 = 0
+ echo 'using new PROJ version 5+ API'
using new PROJ version 5+ API
+ echo -n 'checking for location of External PROJ library... '
checking for location of External PROJ library... + echo 'configure:5889: checking for location of External PROJ library'
+ case "$with_proj_libs" in
+ echo C:/OSGeo4W64/lib
C:/OSGeo4W64/lib
+ test -n C:/OSGeo4W64/lib
+ for dir in '$with_proj_libs'
+ test -d C:/OSGeo4W64/lib
+ PROJLIB=' -LC:/OSGeo4W64/lib'
+ ac_save_ldflags=' -Wl,--export-dynamic,--enable-runtime-pseudo-reloc  -LC:\\OSGEO4~1/lib'
+ LDFLAGS=' -LC:/OSGeo4W64/lib  -Wl,--export-dynamic,--enable-runtime-pseudo-reloc  -LC:\\OSGEO4~1/lib'
+ echo -n 'checking for proj_pj_info in -lproj... '
checking for proj_pj_info in -lproj... + echo 'configure:5913: checking for proj_pj_info in -lproj'
++ echo proj_proj_pj_info
++ sed y%./+-%__p_%
+ ac_lib_var=proj_proj_pj_info
+ ac_save_LIBS=
+ LIBS='-lproj  '
+ cat
+ eval echo configure:5930: '"${CC-cc}' -o 'conftest${ac_exeext}' '$CFLAGS' '$CPPFLAGS' '$LDFLAGS' 'conftest.$ac_ext' '$LIBS' '1>&5"'
++ echo configure:5930: 'gcc -o conftest.exe -g -O2    -IC:/OSGeo4W64/include  -LC:/OSGeo4W64/lib  -Wl,--export-dynamic,--enable-runtime-pseudo-reloc  -LC:\\OSGEO4~1/lib conftest.c -lproj   1>&5'
+ echo 'configure: failed program was:'
+ cat conftest.c
+ rm -rf conftest.c
+ eval ac_cv_lib_proj_proj_pj_info=no
++ ac_cv_lib_proj_proj_pj_info=no
+ rm -f 'conftest*'
+ LIBS=
++ echo '$ac_cv_lib_proj_proj_pj_info'
+ eval 'test "$ac_cv_lib_proj_proj_pj_info" = yes'
++ test no = yes
+ echo no
no
+ LDFLAGS=' -Wl,--export-dynamic,--enable-runtime-pseudo-reloc  -LC:\\OSGEO4~1/lib'
+ echo 'configure: error: *** Unable to locate External PROJ library.'
configure: error: *** Unable to locate External PROJ library.
+ exit 1
```

Reply:
```
C:\OSGeo4W64\lib>ls | grep proj
proj_5_2.lib
proj_i.lib
```

Reply:
with fec7cb9 it seems configuration seems to work according to configure.log and proj seems to be detected:

```
configure:5563: checking for zstd.h
configure:5571: gcc -E    -IC:/OSGeo4W64/include conftest.c >/dev/null 2>conftest.out
+ eval '$CPP' '$CPPFLAGS' conftest.c '>/dev/null' '2>conftest.out'
++ gcc -E -IC:/OSGeo4W64/include conftest.c
configure:5605: checking for location of zstd library
configure:5630: checking for ZSTD_compress in -lzstd
configure:5647: gcc -o conftest.exe -g -O2    -IC:/OSGeo4W64/include   -Wl,--export-dynamic,--enable-runtime-pseudo-reloc  -LC:\\OSGEO4~1/lib conftest.c -lzstd   1>&5
+ eval '${CC-cc}' -o 'conftest${ac_exeext}' '$CFLAGS' '$CPPFLAGS' '$LDFLAGS' 'conftest.$ac_ext' '$LIBS' '1>&5'
++ gcc -o conftest.exe -g -O2 -IC:/OSGeo4W64/include -Wl,--export-dynamic,--enable-runtime-pseudo-reloc '-LC:\\OSGEO4~1/lib' conftest.c -lzstd
c:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: warning: --export-dynamic is not supported for PE+ targets, did you mean --export-all-symbols?
configure:5698: checking for location of External PROJ includes
configure:5725: checking for proj.h
configure:5733: gcc -E  -IC:/OSGeo4W64/include   -IC:/OSGeo4W64/include conftest.c >/dev/null 2>conftest.out
+ eval '$CPP' '$CPPFLAGS' conftest.c '>/dev/null' '2>conftest.out'
++ gcc -E -IC:/OSGeo4W64/include -IC:/OSGeo4W64/include conftest.c
configure:5765: checking External PROJ major version
configure:5785: gcc -o conftest.exe -g -O2   -IC:/OSGeo4W64/include   -IC:/OSGeo4W64/include  -Wl,--export-dynamic,--enable-runtime-pseudo-reloc  -LC:\\OSGEO4~1/lib conftest.c  1>&5
+ eval '${CC-cc}' -o 'conftest${ac_exeext}' '$CFLAGS' '$CPPFLAGS' '$LDFLAGS' 'conftest.$ac_ext' '$LIBS' '1>&5'
++ gcc -o conftest.exe -g -O2 -IC:/OSGeo4W64/include -IC:/OSGeo4W64/include -Wl,--export-dynamic,--enable-runtime-pseudo-reloc '-LC:\\OSGEO4~1/lib' conftest.c
c:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: warning: --export-dynamic is not supported for PE+ targets, did you mean --export-all-symbols?
configure:5802: checking External PROJ minor version
configure:5822: gcc -o conftest.exe -g -O2   -IC:/OSGeo4W64/include   -IC:/OSGeo4W64/include  -Wl,--export-dynamic,--enable-runtime-pseudo-reloc  -LC:\\OSGEO4~1/lib conftest.c  1>&5
+ eval '${CC-cc}' -o 'conftest${ac_exeext}' '$CFLAGS' '$CPPFLAGS' '$LDFLAGS' 'conftest.$ac_ext' '$LIBS' '1>&5'
++ gcc -o conftest.exe -g -O2 -IC:/OSGeo4W64/include -IC:/OSGeo4W64/include -Wl,--export-dynamic,--enable-runtime-pseudo-reloc '-LC:\\OSGEO4~1/lib' conftest.c
c:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: warning: --export-dynamic is not supported for PE+ targets, did you mean --export-all-symbols?
configure:5839: checking External PROJ patch version
configure:5859: gcc -o conftest.exe -g -O2   -IC:/OSGeo4W64/include   -IC:/OSGeo4W64/include  -Wl,--export-dynamic,--enable-runtime-pseudo-reloc  -LC:\\OSGEO4~1/lib conftest.c  1>&5
+ eval '${CC-cc}' -o 'conftest${ac_exeext}' '$CFLAGS' '$CPPFLAGS' '$LDFLAGS' 'conftest.$ac_ext' '$LIBS' '1>&5'
++ gcc -o conftest.exe -g -O2 -IC:/OSGeo4W64/include -IC:/OSGeo4W64/include -Wl,--export-dynamic,--enable-runtime-pseudo-reloc '-LC:\\OSGEO4~1/lib' conftest.c
c:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: warning: --export-dynamic is not supported for PE+ targets, did you mean --export-all-symbols?
configure:5889: checking for location of External PROJ library
configure:5913: checking for proj_pj_info in -lproj
configure:5930: gcc -o conftest.exe -g -O2    -IC:/OSGeo4W64/include  -LC:/msys64/usr/src/grass_win_fix/grassw/mswindows/osgeo4w/lib  -Wl,--export-dynamic,--enable-runtime-pseudo-reloc  -LC:\\OSGEO4~1/lib conftest.c -lproj   1>&5
+ eval '${CC-cc}' -o 'conftest${ac_exeext}' '$CFLAGS' '$CPPFLAGS' '$LDFLAGS' 'conftest.$ac_ext' '$LIBS' '1>&5'
++ gcc -o conftest.exe -g -O2 -IC:/OSGeo4W64/include -LC:/msys64/usr/src/grass_win_fix/grassw/mswindows/osgeo4w/lib -Wl,--export-dynamic,--enable-runtime-pseudo-reloc '-LC:\\OSGEO4~1/lib' conftest.c -lproj
c:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: warning: --export-dynamic is not supported for PE+ targets, did you mean --export-all-symbols?
configure:6138: checking for location of External PROJ data files
configure:6169: checking for C:/OSGeo4W64/share/proj/epsg
configure:6202: checking whether to use regex
configure:6222: checking for location of regex includes
configure:6248: checking for regex.h
```

```
+ case "$ac_option" in
++ echo --with-proj-share=C:/OSGeo4W64/share/proj
++ sed 's/[-_a-zA-Z0-9]*=//'
+ ac_optarg=C:/OSGeo4W64/share/proj
+ case "$ac_option" in
++ echo --with-proj-share=C:/OSGeo4W64/share/proj
++ sed -e 's/-*with-//' -e 's/=.*//'
+ ac_package=proj-share
++ echo proj-share
++ sed 's/[-_a-zA-Z0-9]//g'
+ test -n ''
++ echo proj-share
++ sed s/-/_/g
+ ac_package=proj_share
+ case "$ac_option" in
+ eval 'with_proj_share='\''C:/OSGeo4W64/share/proj'\'''
++ with_proj_share=C:/OSGeo4W64/share/proj
+ for ac_option in '"$@"'
+ test -n ''
+ case "$ac_option" in
++ echo --with-proj-includes=C:/OSGeo4W64/include
++ sed 's/[-_a-zA-Z0-9]*=//'
+ ac_optarg=C:/OSGeo4W64/include
+ case "$ac_option" in
++ echo --with-proj-includes=C:/OSGeo4W64/include
++ sed -e 's/-*with-//' -e 's/=.*//'
+ ac_package=proj-includes
++ echo proj-includes
++ sed 's/[-_a-zA-Z0-9]//g'
+ test -n ''
++ echo proj-includes
++ sed s/-/_/g
+ ac_package=proj_includes
+ case "$ac_option" in
+ eval 'with_proj_includes='\''C:/OSGeo4W64/include'\'''
++ with_proj_includes=C:/OSGeo4W64/include
+ for ac_option in '"$@"'
+ test -n ''
+ case "$ac_option" in
++ echo --with-proj-libs=C:/msys64/usr/src/grass_win_fix/grassw/mswindows/osgeo4w/lib
++ sed 's/[-_a-zA-Z0-9]*=//'
+ ac_optarg=C:/msys64/usr/src/grass_win_fix/grassw/mswindows/osgeo4w/lib
+ case "$ac_option" in
++ echo --with-proj-libs=C:/msys64/usr/src/grass_win_fix/grassw/mswindows/osgeo4w/lib
++ sed -e 's/-*with-//' -e 's/=.*//'
+ ac_package=proj-libs
++ echo proj-libs
++ sed 's/[-_a-zA-Z0-9]//g'
+ test -n ''
++ echo proj-libs
++ sed s/-/_/g
+ ac_package=proj_libs
+ case "$ac_option" in
+ eval 'with_proj_libs='\''C:/msys64/usr/src/grass_win_fix/grassw/mswindows/osgeo4w/lib'\'''
++ with_proj_libs=C:/msys64/usr/src/grass_win_fix/grassw/mswindows/osgeo4w/lib
```

compilation started

```
GRASS GIS 7.8.1dev b23b78fac compilation log
--------------------------------------------------
Started compilation: Sat Oct 12 10:40:53 GMT 2019
--
Errors in:
```
but compilation seems to hang around:

```
/usr/bin/install -c  -m 644 Make/ShScriptRules.make /c/msys64/usr/src/grass_win_fix/grassw/dist.x86_64-w64-mingw32/include/Make/ShScriptRules.make
/usr/bin/install -c  -m 644 Make/Shlib.make /c/msys64/usr/src/grass_win_fix/grassw/dist.x86_64-w64-mingw32/include/Make/Shlib.make
/usr/bin/install -c  -m 644 Make/Sphinx.make /c/msys64/usr/src/grass_win_fix/grassw/dist.x86_64-w64-mingw32/include/Make/Sphinx.make
/usr/bin/install -c  -m 644 Make/Stlib.make /c/msys64/usr/src/grass_win_fix/grassw/dist.x86_64-w64-mingw32/include/Make/Stlib.make
/usr/bin/install -c  -m 644 Make/Vars.make /c/msys64/usr/src/grass_win_fix/grassw/dist.x86_64-w64-mingw32/include/Make/Vars.make
make[2]: Leaving directory `/c/msys64/usr/src/grass_win_fix/grassw/include'
make[2]: Entering directory `/c/msys64/usr/src/grass_win_fix/grassw/lib'
make[3]: Entering directory `/c/msys64/usr/src/grass_win_fix/grassw/lib/datetime'
test -d OBJ.x86_64-w64-mingw32 || mkdir -p OBJ.x86_64-w64-mingw32
gcc -IC:/OSGeo4W64/include -g -O2   -IC:/OSGeo4W64/include -I/c/msys64/usr/src/grass_win_fix/grassw/dist.x86_64-w64-mingw32/include -I/c/msys64/usr/src/grass_win_fix/grassw/dist.x86_64-w64-mingw32/include  -D_FILE_OFFSET_BITS=64  -DPACKAGE=\""grasslibs"\"   -I/c/msys64/usr/src/grass_win_fix/grassw/dist.x86_64-w64-mingw32/include -I/c/msys64/usr/src/grass_win_fix/grassw/dist.x86_64-w64-mingw32/include -DRELDIR=\"lib/datetime\" -o OBJ.x86_64-w64-mingw32/between.o -c between.c
gcc -IC:/OSGeo4W64/include -g -O2   -IC:/OSGeo4W64/include -I/c/msys64/usr/src/grass_win_fix/grassw/dist.x86_64-w64-mingw32/include -I/c/msys64/usr/src/grass_win_fix/grassw/dist.x86_64-w64-mingw32/include  -D_FILE_OFFSET_BITS=64  -DPACKAGE=\""grasslibs"\"   -I/c/msys64/usr/src/grass_win_fix/grassw/dist.x86_64-w64-mingw32/include -I/c/msys64/usr/src/grass_win_fix/grassw/dist.x86_64-w64-mingw32/include -DRELDIR=\"lib/datetime\" -o OBJ.x86_64-w64-mingw32/change.o -c change.c
gcc -IC:/OSGeo4W64/include -g -O2   -IC:/OSGeo4W64/include -I/c/msys64/usr/src/grass_win_fix/grassw/dist.x86_64-w64-mingw32/include -I/c/msys64/usr/src/grass_win_fix/grassw/dist.x86_64-w64-mingw32/include  -D_FILE_OFFSET_BITS=64  -DPACKAGE=\""grasslibs"\"   -I/c/msys64/usr/src/grass_win_fix/grassw/dist.x86_64-w64-mingw32/include -I/c/msys64/usr/src/grass_win_fix/grassw/dist.x86_64-w64-mingw32/include -DRELDIR=\"lib/datetime\" -o OBJ.x86_64-w64-mingw32/copy.o -c copy.c
gcc -IC:/OSGeo4W64/include -g -O2   -IC:/OSGeo4W64/include -I/c/msys64/usr/src/grass_win_fix/grassw/dist.x86_64-w64-mingw32/include -I/c/msys64/usr/src/grass_win_fix/grassw/dist.x86_64-w64-mingw32/include  -D_FILE_OFFSET_BITS=64  -DPACKAGE=\""grasslibs"\"   -I/c/msys64/usr/src/grass_win_fix/grassw/dist.x86_64-w64-mingw32/include -I/c/msys64/usr/src/grass_win_fix/grassw/dist.x86_64-w64-mingw32/include -DRELDIR=\"lib/datetime\" -o OBJ.x86_64-w64-mingw32/diff.o -c diff.c
```

Reply:
ok, now did package.sh without bash -x, then there is still the same error message:

```
hkmyr@DESKTOP-VADT8Q4 MINGW64 /usr/src/grass_win_fix/grass
$ OSGEO4W_POSTFIX=64 ./mswindows/osgeo4w/package.sh
`C:/msys64/mingw64/bin/zlib1.dll' -> `dist.x86_64-w64-mingw32/bin/zlib1.dll'
`C:/msys64/mingw64/bin/libbz2-1.dll' -> `dist.x86_64-w64-mingw32/bin/libbz2-1.dll'
`C:/msys64/mingw64/bin/libiconv-2.dll' -> `dist.x86_64-w64-mingw32/bin/libiconv-2.dll'
`C:/msys64/mingw64/bin/libexpat-1.dll' -> `dist.x86_64-w64-mingw32/bin/libexpat-1.dll'
`C:/msys64/mingw64/bin/libfontconfig-1.dll' -> `dist.x86_64-w64-mingw32/bin/libfontconfig-1.dll'
`C:/msys64/mingw64/bin/libintl-8.dll' -> `dist.x86_64-w64-mingw32/bin/libintl-8.dll'
`C:/msys64/mingw64/bin/libsystre-0.dll' -> `dist.x86_64-w64-mingw32/bin/libsystre-0.dll'
`C:/msys64/mingw64/bin/libtre-5.dll' -> `dist.x86_64-w64-mingw32/bin/libtre-5.dll'
`C:/msys64/mingw64/bin/libwinpthread-1.dll' -> `dist.x86_64-w64-mingw32/bin/libwinpthread-1.dll'
`C:/msys64/mingw64/bin/libcairo-2.dll' -> `dist.x86_64-w64-mingw32/bin/libcairo-2.dll'
`C:/msys64/mingw64/bin/libpixman-1-0.dll' -> `dist.x86_64-w64-mingw32/bin/libpixman-1-0.dll'
`C:/msys64/mingw64/bin/libpng16-16.dll' -> `dist.x86_64-w64-mingw32/bin/libpng16-16.dll'
`C:/msys64/mingw64/bin/libfreetype-6.dll' -> `dist.x86_64-w64-mingw32/bin/libfreetype-6.dll'
`C:/msys64/mingw64/bin/libharfbuzz-0.dll' -> `dist.x86_64-w64-mingw32/bin/libharfbuzz-0.dll'
`C:/msys64/mingw64/bin/libglib-2.0-0.dll' -> `dist.x86_64-w64-mingw32/bin/libglib-2.0-0.dll'
`C:/msys64/mingw64/bin/libgraphite2.dll' -> `dist.x86_64-w64-mingw32/bin/libgraphite2.dll'
`C:/msys64/mingw64/bin/libpcre-1.dll' -> `dist.x86_64-w64-mingw32/bin/libpcre-1.dll'
`C:/msys64/mingw64/bin/libstdc++-6.dll' -> `dist.x86_64-w64-mingw32/bin/libstdc++-6.dll'
`C:/msys64/mingw64/bin/libgcc_s_seh-1.dll' -> `dist.x86_64-w64-mingw32/bin/libgcc_s_seh-1.dll'
`C:/OSGeo4W64/lib/libpq.lib' -> `mswindows/osgeo4w/lib/pq.lib'
`C:/OSGeo4W64/lib/proj_i.lib' -> `mswindows/osgeo4w/lib/proj.lib'
Sat Oct 12 12:05:32 GMT 2019: STARTING remove old logs
Sat Oct 12 12:05:33 GMT 2019: FINISHED remove old logs AFTER 1s
Sat Oct 12 12:05:33 GMT 2019: STARTING configure
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
loading site script /mingw64/etc/config.site
checking host system type... config.sub: missing argument
Try `config.sub --help' for more information.

checking for gcc... gcc
checking whether the C compiler (gcc  ) works... yes
checking whether the C compiler (gcc  ) is a cross-compiler... no
checking whether we are using GNU C... yes
checking whether gcc accepts -g... yes
checking for Cygwin environment... no
checking for mingw32 environment... yes
checking for executable suffix... .exe
checking for full floating-point support... yes
checking for pwd... /C/OSGEO4~1/apps/msys/bin/pwd
checking for source directory... /c/msys64/usr/src/grass_win_fix/grass
checking for build directory... /c/msys64/usr/src/grass_win_fix/grass
checking for git... /C/msys64/usr/bin/git
checking for MacOSX App... no
checking for MacOSX architectures... no
checking for MacOSX SDK... no
checking how to build libraries... shared
configure: error: ***Unknown platform: ***
```

Reply:
now added _#!/usr/bin/bash -x_ in package.sh:

```
$ OSGEO4W_POSTFIX=64 ./mswindows/osgeo4w/package.sh
+ set -e
++ pwd
+ PWD=/usr/src/grass_win_fix/grass
+ '[' -d mswindows ']'
+ mkdir -p /tmp
+ '[' -d /tmp ']'
+ '[' -z ']'
+ PACKAGE_PATCH=1
+ '[' -z ']'
+ PACKAGE_POSTFIX=
+ '[' -z 64 ']'
+ '[' 64 = 64 ']'
+ MINGW_POSTFIX=64
+ export OSGEO4W_ROOT_MSYS=/c/OSGeo4W64
+ OSGEO4W_ROOT_MSYS=/c/OSGeo4W64
++ cygpath -w /c/OSGeo4W64
+ export 'OSGEO4W_ROOT=C:\OSGeo4W64'
+ OSGEO4W_ROOT='C:\OSGeo4W64'
+ fetchenv /c/OSGeo4W64/bin/o4w_env.bat
+ local IFS
+ IFS=
+ batch=/c/OSGeo4W64/bin/o4w_env.bat
+ shift
++ mktemp /tmp/srcenv.XXXXXXXXXX
+ srcenv=/tmp/srcenv.th7Og1ZoI5
++ mktemp /tmp/dstenv.XXXXXXXXXX
+ dstenv=/tmp/dstenv.V8Uu4YfSip
++ mktemp /tmp/diffenv.XXXXXXXXXX
+ diffenv=/tmp/diffenv.bIVXLP9FWL
+ args=
+ cmd.exe //c set
++ cygpath -w /c/OSGeo4W64/bin/o4w_env.bat
+ cmd.exe //c 'call C:\OSGeo4W64\bin\o4w_env.bat  \>nul 2\>nul \& set'
+ diff -u /tmp/srcenv.th7Og1ZoI5 /tmp/dstenv.V8Uu4YfSip
+ sed -f mswindows/osgeo4w/envdiff.sed
+ . /tmp/diffenv.bIVXLP9FWL
++ export CYGWIN=nodosfilewarning
++ CYGWIN=nodosfilewarning
++ export 'GDAL_DATA=C:\\OSGEO4~1\\share\\gdal'
++ GDAL_DATA='C:\\OSGEO4~1\\share\\gdal'
++ export 'GDAL_DRIVER_PATH=C:\\OSGEO4~1\\bin\\gdalplugins'
++ GDAL_DRIVER_PATH='C:\\OSGEO4~1\\bin\\gdalplugins'
++ export 'GEOTIFF_CSV=C:\\OSGEO4~1\\share\\epsg_csv'
++ GEOTIFF_CSV='C:\\OSGEO4~1\\share\\epsg_csv'
++ export JPEGMEM=1000000
++ JPEGMEM=1000000
++ export O4W_QT_BINARIES=C:/OSGEO4~1/bin
++ O4W_QT_BINARIES=C:/OSGEO4~1/bin
++ export O4W_QT_HEADERS=C:/OSGEO4~1/include/qt4
++ O4W_QT_HEADERS=C:/OSGEO4~1/include/qt4
++ export O4W_QT_LIBRARIES=C:/OSGEO4~1/lib
++ O4W_QT_LIBRARIES=C:/OSGEO4~1/lib
++ export O4W_QT_PLUGINS=C:/OSGEO4~1/apps/qt4/plugins
++ O4W_QT_PLUGINS=C:/OSGEO4~1/apps/qt4/plugins
++ export O4W_QT_PREFIX=C:/OSGEO4~1/apps/qt4
++ O4W_QT_PREFIX=C:/OSGEO4~1/apps/qt4
++ export O4W_QT_TRANSLATIONS=C:/OSGEO4~1/apps/qt4/translations
++ O4W_QT_TRANSLATIONS=C:/OSGEO4~1/apps/qt4/translations
++ export 'OSGEO4W_ROOT=C:\\OSGEO4~1'
++ OSGEO4W_ROOT='C:\\OSGEO4~1'
++ export 'PATH={app}:/C/OSGEO4~1/apps/Python27/Scripts:/C/OSGEO4~1/bin:/C/WINDOWS/system32:/C/WINDOWS:/C/WINDOWS/system32/WBem:/C/OSGEO4~1/apps/msys/bin:/C/Program Files/R/R-3.6.0/bin/x64'
++ PATH='{app}:/C/OSGEO4~1/apps/Python27/Scripts:/C/OSGEO4~1/bin:/C/WINDOWS/system32:/C/WINDOWS:/C/WINDOWS/system32/WBem:/C/OSGEO4~1/apps/msys/bin:/C/Program Files/R/R-3.6.0/bin/x64'
++ export 'PROJ_LIB=C:\\OSGEO4~1\\share\\proj'
++ PROJ_LIB='C:\\OSGEO4~1\\share\\proj'
++ export 'PYTHONHOME=C:\\OSGEO4~1\\apps\\Python27'
++ PYTHONHOME='C:\\OSGEO4~1\\apps\\Python27'
++ export 'QT_PLUGIN_PATH=C:\\OSGEO4~1\\apps\\qt4\\plugins'
++ QT_PLUGIN_PATH='C:\\OSGEO4~1\\apps\\qt4\\plugins'
++ export QT_RASTER_CLIP_LIMIT=4096
++ QT_RASTER_CLIP_LIMIT=4096
++ export R_REGISTRY=1
++ R_REGISTRY=1
+ PATH='{app}:/C/OSGEO4~1/apps/Python27/Scripts:/C/OSGEO4~1/bin:/C/WINDOWS/system32:/C/WINDOWS:/C/WINDOWS/system32/WBem:/C/OSGEO4~1/apps/msys/bin:/C/Program Files/R/R-3.6.0/bin/x64:/usr/bin:/mingw64/bin/:/usr/src/grass_win_fix/grass/mswindows/osgeo4w/lib:/usr/src/grass_win_fix/grass/mswindows/osgeo4w:/c/windows32/system32:/c/windows:/c/windows32/system32:/c/windows'
+ rm -f /tmp/srcenv.th7Og1ZoI5 /tmp/dstenv.V8Uu4YfSip /tmp/diffenv.bIVXLP9FWL
+ fetchenv /c/OSGeo4W64/bin/py3_env.bat
+ local IFS
+ IFS=
+ batch=/c/OSGeo4W64/bin/py3_env.bat
+ shift
++ mktemp /tmp/srcenv.XXXXXXXXXX
+ srcenv=/tmp/srcenv.Tbl9Xs6buj
++ mktemp /tmp/dstenv.XXXXXXXXXX
+ dstenv=/tmp/dstenv.yZ37aDKI64
++ mktemp /tmp/diffenv.XXXXXXXXXX
+ diffenv=/tmp/diffenv.FFDrpfv3nt
+ args=
+ cmd.exe //c set
++ cygpath -w /c/OSGeo4W64/bin/py3_env.bat
+ cmd.exe //c 'call C:\OSGeo4W64\bin\py3_env.bat  \>nul 2\>nul \& set'
+ diff -u /tmp/srcenv.Tbl9Xs6buj /tmp/dstenv.yZ37aDKI64
+ sed -f mswindows/osgeo4w/envdiff.sed
+ . /tmp/diffenv.FFDrpfv3nt
++ export 'PATH=/C//OSGEO4~1/apps/Python37:/C//OSGEO4~1/apps/Python37/Scripts:{app}:/C/OSGEO4~1/apps/Python27/Scripts:/C/OSGEO4~1/bin:/C/WINDOWS/system32:/C/WINDOWS:/C/WINDOWS/system32/WBem:/C/OSGEO4~1/apps/msys/bin:/C/Program Files/R/R-3.6.0/bin/x64:/C/msys64/usr/bin:/C/msys64/mingw64/bin/:/C/msys64/usr/src/grass_win_fix/grass/mswindows/osgeo4w/lib:/C/msys64/usr/src/grass_win_fix/grass/mswindows/osgeo4w:/C/windows32/system32:/C/windows:/C/windows32/system32:/C/windows'
++ PATH='/C//OSGEO4~1/apps/Python37:/C//OSGEO4~1/apps/Python37/Scripts:{app}:/C/OSGEO4~1/apps/Python27/Scripts:/C/OSGEO4~1/bin:/C/WINDOWS/system32:/C/WINDOWS:/C/WINDOWS/system32/WBem:/C/OSGEO4~1/apps/msys/bin:/C/Program Files/R/R-3.6.0/bin/x64:/C/msys64/usr/bin:/C/msys64/mingw64/bin/:/C/msys64/usr/src/grass_win_fix/grass/mswindows/osgeo4w/lib:/C/msys64/usr/src/grass_win_fix/grass/mswindows/osgeo4w:/C/windows32/system32:/C/windows:/C/windows32/system32:/C/windows'
++ export 'PYTHONHOME=C:\\\\OSGEO4~1\\apps\\Python37'
++ PYTHONHOME='C:\\\\OSGEO4~1\\apps\\Python37'
+ PATH='/C//OSGEO4~1/apps/Python37:/C//OSGEO4~1/apps/Python37/Scripts:{app}:/C/OSGEO4~1/apps/Python27/Scripts:/C/OSGEO4~1/bin:/C/WINDOWS/system32:/C/WINDOWS:/C/WINDOWS/system32/WBem:/C/OSGEO4~1/apps/msys/bin:/C/Program Files/R/R-3.6.0/bin/x64:/C/msys64/usr/bin:/C/msys64/mingw64/bin/:/C/msys64/usr/src/grass_win_fix/grass/mswindows/osgeo4w/lib:/C/msys64/usr/src/grass_win_fix/grass/mswindows/osgeo4w:/C/windows32/system32:/C/windows:/C/windows32/system32:/C/windows:/usr/bin:/mingw64/bin/:/usr/src/grass_win_fix/grass/mswindows/osgeo4w/lib:/usr/src/grass_win_fix/grass/mswindows/osgeo4w:/c/windows32/system32:/c/windows:/c/windows32/system32:/c/windows'
+ rm -f /tmp/srcenv.Tbl9Xs6buj /tmp/dstenv.yZ37aDKI64 /tmp/diffenv.FFDrpfv3nt
+ PATH='/mingw64/lib/ccache/bin:/C//OSGEO4~1/apps/Python37:/C//OSGEO4~1/apps/Python37/Scripts:{app}:/C/OSGEO4~1/apps/Python27/Scripts:/C/OSGEO4~1/bin:/C/WINDOWS/system32:/C/WINDOWS:/C/WINDOWS/system32/WBem:/C/OSGEO4~1/apps/msys/bin:/C/Program Files/R/R-3.6.0/bin/x64:/C/msys64/usr/bin:/C/msys64/mingw64/bin/:/C/msys64/usr/src/grass_win_fix/grass/mswindows/osgeo4w/lib:/C/msys64/usr/src/grass_win_fix/grass/mswindows/osgeo4w:/C/windows32/system32:/C/windows:/C/windows32/system32:/C/windows:/usr/bin:/mingw64/bin/:/usr/src/grass_win_fix/grass/mswindows/osgeo4w/lib:/usr/src/grass_win_fix/grass/mswindows/osgeo4w:/c/windows32/system32:/c/windows:/c/windows32/system32:/c/windows'
++ date +%s
+ T0=1570882222
+ LT=1570882222
+ CS=
+ exec
+ read MAJOR
+ read MINOR
+ read PATCH
+ export VERSION=7.8.1dev
+ VERSION=7.8.1dev
+ export POSTFIX=78
+ POSTFIX=78
+ [[ 1dev == *svn* ]]
+ GRASS_EXECUTABLE=grass78
+ '[' -f mswindows/osgeo4w/package.log ']'
+ i=0
+ '[' -f mswindows/osgeo4w/package.log.0 ']'
+ mv mswindows/osgeo4w/package.log mswindows/osgeo4w/package.log.0
+ exec
++ tee mswindows/osgeo4w/package.log
+ '[' 64 = 64 ']'
+ mingw_libgcc=libgcc_s_seh-1.dll
+ DLLS='/mingw64/bin/zlib1.dll
        /mingw64/bin/libbz2-1.dll
        /mingw64/bin/libiconv-2.dll
        /mingw64/bin/libexpat-1.dll
        /mingw64/bin/libfontconfig-1.dll
        /mingw64/bin/libintl-8.dll
        /mingw64/bin/libsystre-0.dll
        /mingw64/bin/libtre-5.dll
        /mingw64/bin/libwinpthread-1.dll
        /mingw64/bin/libcairo-2.dll
        /mingw64/bin/libpixman-1-0.dll
        /mingw64/bin/libpng16-16.dll
        /mingw64/bin/libfreetype-6.dll
        /mingw64/bin/libharfbuzz-0.dll
        /mingw64/bin/libglib-2.0-0.dll
        /mingw64/bin/libgraphite2.dll
        /mingw64/bin/libpcre-1.dll
        /mingw64/bin/libstdc++-6.dll
        /mingw64/bin/libgcc_s_seh-1.dll'
+ '[' 64 = 64 ']'
+ conf_host=x86_64-w64-mingw32
+ conf_opts=--with-liblas=/usr/src/grass_win_fix/grass/mswindows/osgeo4w/liblas-config
+ mkdir -p dist.x86_64-w64-mingw32/bin
+ cp -uv /mingw64/bin/zlib1.dll /mingw64/bin/libbz2-1.dll /mingw64/bin/libiconv-2.dll /mingw64/bin/libexpat-1.dll /mingw64/bin/libfontconfig-1.dll /mingw64/bin/libintl-8.dll /mingw64/bin/libsystre-0.dll /mingw64/bin/libtre-5.dll /mingw64/bin/libwinpthread-1.dll /mingw64/bin/libcairo-2.dll /mingw64/bin/libpixman-1-0.dll /mingw64/bin/libpng16-16.dll /mingw64/bin/libfreetype-6.dll /mingw64/bin/libharfbuzz-0.dll /mingw64/bin/libglib-2.0-0.dll /mingw64/bin/libgraphite2.dll /mingw64/bin/libpcre-1.dll /mingw64/bin/libstdc++-6.dll /mingw64/bin/libgcc_s_seh-1.dll dist.x86_64-w64-mingw32/bin
+ mkdir -p mswindows/osgeo4w/lib
+ cp -uv /c/OSGeo4W64/lib/libpq.lib mswindows/osgeo4w/lib/pq.lib
+ cp -uv /c/OSGeo4W64/lib/proj_i.lib mswindows/osgeo4w/lib/proj.lib
+ '[' -f mswindows/osgeo4w/configure-stamp ']'
+ '[' -e include/Make/Platform.make ']'
+ log remove old logs
+ local D T
++ date
+ NOW='Sat Oct 12 12:10:23 GMT 2019'
++ date +%s
+ T=1570882223
+ '[' -n '' ']'
+ CS='remove old logs'
+ LT=1570882223
+ '[' -n 'remove old logs' ']'
+ echo Sat Oct 12 12:10:23 GMT 2019: STARTING remove old logs
Sat Oct 12 12:10:23 GMT 2019: STARTING remove old logs
+ rm -f 'mswindows/osgeo4w/package.log.[0-9][0-9][0-9]'
+ log configure
+ local D T
++ date
+ NOW='Sat Oct 12 12:10:23 GMT 2019'
++ date +%s
+ T=1570882223
+ '[' -n 'remove old logs' ']'
+ local D H M S
+ S=0
+ M=0
+ S=0
+ H=0
+ M=0
+ D=0
+ H=0
+ echo -n 'Sat Oct 12 12:10:23 GMT 2019: FINISHED remove old logs AFTER '
Sat Oct 12 12:10:23 GMT 2019: FINISHED remove old logs AFTER + ((  D>0  ))
+ ((  H>0  ))
+ ((  M>0  ))
+ echo 0s
0s
+ CS=configure
+ LT=1570882223
+ '[' -n configure ']'
+ echo Sat Oct 12 12:10:23 GMT 2019: STARTING configure
Sat Oct 12 12:10:23 GMT 2019: STARTING configure
+ ./configure --host=x86_64-w64-mingw32 '--with-libs=C:\\OSGEO4~1/lib' --with-includes=/c/OSGeo4W64/include --libexecdir=/c/OSGeo4W64/bin --prefix=/c/OSGeo4W64/apps/grass --bindir=/c/OSGeo4W64/bin --includedir=/c/OSGeo4W64/include --without-x --with-cxx --enable-shared --enable-largefile --with-fftw --with-freetype --with-freetype-includes=/mingw64/include/freetype2 --with-proj-share=/c/OSGeo4W64/share/proj --with-proj-includes=/c/OSGeo4W64/include --with-proj-libs=/usr/src/grass_win_fix/grass/mswindows/osgeo4w/lib --with-postgres --with-postgres-includes=/c/OSGeo4W64/include --with-postgres-libs=/usr/src/grass_win_fix/grass/mswindows/osgeo4w/lib --with-gdal=/usr/src/grass_win_fix/grass/mswindows/osgeo4w/gdal-config --with-geos=/usr/src/grass_win_fix/grass/mswindows/osgeo4w/geos-config --with-sqlite --with-regex --with-nls --with-zstd --with-odbc --with-cairo --with-opengl=windows --with-bzlib --with-liblas=/usr/src/grass_win_fix/grass/mswindows/osgeo4w/liblas-config
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
/usr/bin/sed: -e expression #1, char 18: unterminated `s' command
loading site script /mingw64/etc/config.site
checking host system type... config.sub: missing argument
Try `config.sub --help' for more information.

checking for gcc... gcc
checking whether the C compiler (gcc  ) works... yes
checking whether the C compiler (gcc  ) is a cross-compiler... no
checking whether we are using GNU C... yes
checking whether gcc accepts -g... yes
checking for Cygwin environment... no
checking for mingw32 environment... yes
checking for executable suffix... .exe
checking for full floating-point support... yes
checking for pwd... /C/OSGEO4~1/apps/msys/bin/pwd
checking for source directory... /c/msys64/usr/src/grass_win_fix/grass
checking for build directory... /c/msys64/usr/src/grass_win_fix/grass
checking for git... /C/msys64/usr/bin/git
checking for MacOSX App... no
checking for MacOSX architectures... no
checking for MacOSX SDK... no
checking how to build libraries... shared
configure: error: ***Unknown platform: ***
```

Reply:
extracting configure options form above

```
./configure 
--host=x86_64-w64-mingw32 
'--with-libs=C:\\OSGEO4~1/lib' 
--with-includes=/c/OSGeo4W64/include 
--libexecdir=/c/OSGeo4W64/bin 
--prefix=/c/OSGeo4W64/apps/grass 
--bindir=/c/OSGeo4W64/bin 
--includedir=/c/OSGeo4W64/include 
--without-x 
--with-cxx 
--enable-shared 
--enable-largefile 
--with-fftw 
--with-freetype 
--with-freetype-includes=/mingw64/include/freetype2 
--with-proj-share=/c/OSGeo4W64/share/proj 
--with-proj-includes=/c/OSGeo4W64/include 
--with-proj-libs=/usr/src/grass_win_fix/grass/mswindows/osgeo4w/lib 
--with-postgres 
--with-postgres-includes=/c/OSGeo4W64/include 
--with-postgres-libs=/usr/src/grass_win_fix/grass/mswindows/osgeo4w/lib 
--with-gdal=/usr/src/grass_win_fix/grass/mswindows/osgeo4w/gdal-config 
--with-geos=/usr/src/grass_win_fix/grass/mswindows/osgeo4w/geos-config 
--with-sqlite 
--with-regex 
--with-nls 
--with-zstd 
--with-odbc 
--with-cairo 
--with-opengl=windows 
--with-bzlib 
--with-liblas=/usr/src/grass_win_fix/grass/mswindows/osgeo4w/liblas-config
```
```
loading site script /mingw64/etc/config.site
checking host system type... config.sub: missing argument
Try `config.sub --help' for more information.
```
looking at _cmd.exe //c set_: 

```
MINGW_CHOST=x86_64-w64-mingw32
MINGW_PACKAGE_PREFIX=mingw-w64-x86_64
```
in C:\msys64\mingw64\etc\config.site

```
# This file is in public domain.
# Original author: Karlson2k (Evgeny Grin)
# Written for MSys2/MinGW64 to help running 'configure' scripts

# Defaults for MinGW64-targeted programs

# Set proper selfname on bash and fallback to default name on other shells
test -n "${BASH_SOURCE}" 2>/dev/null && config_site_me="${BASH_SOURCE[0]##*/}" || config_site_me=config.site

# Set default 'host' to speedup configure
if test -z "$build_alias"; then
  build_alias="${MSYSTEM_CHOST-x86_64-w64-mingw32}" && \
    ${as_echo-echo} "$config_site_me:${as_lineno-$LINENO}: default build_alias set to $build_alias" >&5
fi

# Set default 'prefix' to "/mingw64"
if ( test -z "$prefix" || test "x$prefix" = "xNONE" ) && \
   ( test -z "$exec_prefix" || test "x$exec_prefix" = "xNONE" ); then
  prefix="${MSYSTEM_PREFIX-/mingw64}" && \
    ${as_echo-echo} "$config_site_me:${as_lineno-$LINENO}: default prefix set to $prefix" >&5
fi
```